cloudsync/tasks.py

```diff
@@ -95,14 +132,36 @@ def transcode_from_s3(self, video_id):  # pylint: disable=unused-argument
 
     # Generate thumbnails for the 1st encoding (no point in doing so for each).
     outputs[0]['ThumbnailPattern'] = THUMBNAIL_PATTERN.format(video_file.s3_basename)
-    transcoder = Transcoder(
+    transcoder = TechTVTranscoder(
         settings.ET_PIPELINE_ID,
         settings.AWS_REGION,
         settings.AWS_ACCESS_KEY_ID,
         settings.AWS_SECRET_ACCESS_KEY
     )
-    transcoder.encode(video_input, outputs, Playlists=playlists)
-    transcoder.create_job_for_object(video)
+    try:
+        transcoder.encode(video_input, outputs, Playlists=playlists)
+    except ClientError as e:
+        log.exception('Transcode job creation failed for video %s: %s', video_id, e.response)
+        update_video_status(video, VideoStatus.TRANSCODE_FAILED)
+        self.update_state(task_id=task_id, state=states.FAILURE)
+        transcoder.message = e.response
+        raise e
+    finally:
+        job = transcoder.create_job_for_object(video)
+
+    # Wait for the transcoding job to complete
+    waiter = transcoder.client.get_waiter('job_complete')
+    try:
+        waiter.wait(Id=transcoder.message['Job']['Id'])
```

Comment:
what you described is locking the celery worker.
we probably should skip the execution and enqueue a new task with a delay: something like this
http://docs.celeryproject.org/en/latest/userguide/calling.html#eta-and-countdown